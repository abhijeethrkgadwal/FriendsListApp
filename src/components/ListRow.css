.item{
  display: flex;
  height: 60px;
  flex-direction: row;
  justify-content: space-between;
  border-bottom: 1px solid #c3bcbc;
}

.item-name{
  margin-left: 10px;
}

.item-name h4 span{
  font-size: 10px;
  display: block;
  font-weight: 400;
}

.item-actions{
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-right: 10px;
}

.item-actions button{
  height: 60%;
  margin-right: 10px;
  padding: 8px;
  width: 40px;
  cursor: pointer;
}