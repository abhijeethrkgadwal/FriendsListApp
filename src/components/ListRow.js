import "./ListRow.css"

function ListRow(props) {

  const onDelete = () => {
    let isConfirm = window.confirm('Are you sure want to delete');
    if (isConfirm) {
      props.deleteRow(props.id)
    }
  }

  return (
    <div className="item" data-testid="listRowhead">
      <div className="item-name" data-testid="listRowLabel">
        <h4>{props.name}<span>is your freind</span></h4>
      </div>
      <div className="item-actions" data-testid="listRowButton">
        <button onClick={() => props.makeFavourite(props.id)}>
          {
            props.isFavourite
              ? <i className="fas fa-star"></i>
              : <i className="far fa-star"></i>
          }
        </button>
        <button onClick={onDelete}>
          <i className="fas fa-trash-alt"></i>
        </button>
      </div>
    </div>
  )
}

export default ListRow