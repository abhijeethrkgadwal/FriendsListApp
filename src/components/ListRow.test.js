import { render, cleanup, screen } from "@testing-library/react";
import ListRow from "./ListRow";
import renderer from "react-test-renderer";

afterEach(() => {
  cleanup();
});

test("should render ListRow", () => {
  render(<ListRow />);
  const listRowEle = screen.getByTestId("listRowLabel");
  expect(listRowEle).toBeInTheDocument();
  expect(listRowEle).toContainHTML("h4");
  expect(listRowEle).toContainHTML("span");
});

test("should render ListRow button", () => {
  render(<ListRow />);
  const listRowEleButton = screen.getByTestId("listRowButton");
  expect(listRowEleButton).toBeInTheDocument();
  expect(listRowEleButton).toContainHTML("button");
  expect(listRowEleButton).toContainHTML("i");
});

test("should match ListRow Snapshot", () => {
  const dummyList = { id: 1, name: "Abhi", favourite: false };
  const tree = renderer
    .create(<ListRow id={dummyList.id} name={dummyList.name} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
