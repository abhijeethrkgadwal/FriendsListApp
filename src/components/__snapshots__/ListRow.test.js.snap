// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match ListRow Snapshot 1`] = `
<div
  className="item"
  data-testid="listRowhead"
>
  <div
    className="item-name"
    data-testid="listRowLabel"
  >
    <h4>
      Abhi
      <span>
        is your freind
      </span>
    </h4>
  </div>
  <div
    className="item-actions"
    data-testid="listRowButton"
  >
    <button
      onClick={[Function]}
    >
      <i
        className="far fa-star"
      />
    </button>
    <button
      onClick={[Function]}
    >
      <i
        className="fas fa-trash-alt"
      />
    </button>
  </div>
</div>
`;
